examples/jsm/nodes/utils/MatcapUVNode.js

```diff
@@ -0,0 +1,25 @@
+import TempNode from '../core/TempNode.js';
+import { join, negate, cross, dot, mul, add, transformedNormalView, positionViewDirection } from '../ShaderNode.js';
+
+class MatcapUVNode extends TempNode {
+
+	constructor() {
+
+		super( 'vec2' );
+
+	}
+
+	generate( builder ) {
+
+		const x = join( positionViewDirection.z, 0, negate( positionViewDirection.x ) );
```

Comment:
Does `positionViewDirection` have unit length?

`x` should also have unit length.

Compare with the Matcap shader code.

Reply:
`positionViewDirection` is a normalized vector. I think that I forgot to call `normalize` in `x`

```js
// shader node
const x = normalize( join( positionViewDirection.z, 0, negate( positionViewDirection.x ) ) );

// glsl
vec3 x = normalize( vec3( viewDir.z, 0.0, - viewDir.x ) );
```